.ci/error-prone-check.groovy

```diff
@@ -0,0 +1,342 @@
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.Field
+import groovy.transform.Immutable
+import groovy.xml.XmlUtil
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+@Field final static String SEPARATOR = System.getProperty("file.separator")
+@Field final static Set<String> profiles = Set.of("compile", "test-compile")
+
+final int exitCode
+if (args.length == 2) {
+    exitCode = parseArgumentAndExecute(args[0], args[1])
+}
+else {
+    exitCode = parseArgumentAndExecute(args[0], null)
+}
+System.exit(exitCode)
+
+/**
+ * Parse the command line arguments passed in and execute the branch based on the arguments.
+ * compile - compile the source code of the project.
+ * test-compile - compile the test source code into the test destination directory.
+ *
+ * @param argument command line argument
+ * @return {@code 0} if command executes successfully, {@code 1} otherwise
+ */
+private int parseArgumentAndExecute(String argument, String flag) {
+    final String usageString = "Usage groovy ./.ci/error-prone-check.groovy" +
+            " (compile | test-compile) [-g | --generate-suppression]"
+
+    final int exitCode
+    if (profiles.contains(argument)) {
+        if (flag != null && flag != "-g" && flag != "--generate-suppression") {
+            final String exceptionMessage = "\nUnexpected flag: ${flag}" + usageString
```

Comment:
This needs a newline after `flag`. (and I would delete `\n` at the start)
```
$ groovy error-prone-check.groovy asd
Caught: java.lang.IllegalArgumentException:
Unexpected argument: asdUsage groovy ./.ci/error-prone-check.groovy (compile | test-compile) [-g | --generate-suppression]
java.lang.IllegalArgumentException:
Unexpected argument: asdUsage groovy ./.ci/error-prone-check.groovy (compile | test-compile) [-g | --generate-suppression]
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
...
```

Reply:
Done.